### PR TITLE
Retrieve Discoverable Applications for Sub-Org from the Organization Application Management Service

### PR DESCRIPTION
**Pull request by asha15**
Related PR: https://github.com/wso2-extensions/identity-organization-management/pull/390

Related issue: https://github.com/wso2/product-is/issues/21118

Merge after: https://github.com/wso2-extensions/identity-organization-management/pull/392

## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -491,6 +496,7 @@
         <commons-lang.wso2.version>2.6.0.wso2v1</commons-lang.wso2.version>
         <mavan.findbugsplugin.exclude.file>findbugs-exclude-filter.xml</mavan.findbugsplugin.exclude.file>
         <config.mapper.version>1.0.13</config.mapper.version>
+        <identity.organization.management.version>1.4.45</identity.organization.management.version>
```
Comment (asha15) [inline pom.xml:499]: Bump the version after merging: https://github.com/wso2-extensions/identity-organization-management/pull/392

File: components/org.wso2.carbon.identity.api.user.application/org.wso2.carbon.identity.api.user.application.common/pom.xml [hunk 2]
```diff
@@ -38,5 +38,10 @@
             <artifactId>org.wso2.carbon.identity.application.mgt</artifactId>
             <scope>provided</scope>
         </dependency>
+        <dependency>
+            <groupId>org.wso2.carbon.identity.organization.management</groupId>
+            <artifactId>org.wso2.carbon.identity.organization.management.application</artifactId>
+            <scope>compile</scope>
```
Comment (sadilchamishka) [inline components/org.wso2.carbon.identity.api.user.application/org.wso2.carbon.identity.api.user.application.common/pom.xml:44]: Why the scope should be compile?

Comment (asha15) [reply components/org.wso2.carbon.identity.api.user.application/org.wso2.carbon.identity.api.user.application.common/pom.xml]: Updated the scope to provided

**Review by ShanChathusanda93 — APPROVED** [review]

**Review by sadilchamishka — APPROVED** [review]

File: pom.xml (hunk 1 above)
Comment (asha15) [reply pom.xml]: Fixed with https://github.com/wso2/identity-api-user/pull/202/commits/84eb30c0dde1e831268df350241d464ba6b20f7d

**Review by SujanSanjula96 — APPROVED** [review]

**Pull request merged**